contracts/Strategy.sol

```diff
@@ -198,7 +199,8 @@ contract Strategy is BaseStrategy {
     }
 
     function reduceCollateral(uint256 _amount) internal {
-        require(balanceOfCollateral() >= _amount, "Not enough collateral");
+        // if balanceOfCollateral is not enough, we reduce max amount available
+        if (balanceOfCollateral() >= amount) _amount = balanceOfCollateral();
```

Comment:
when amount is going to be larger than `balanceOfCollateral()`? that would be an issue from the caller.

Reply:
it is an issue from the caller but in any case this would avoid the lock of funds in the strategy (if caller only tries to withdraw full amount for some reason)